### ISSUE TITLE
Runner support for multiple years

### ISSUE THREAD

**Issue by trebinor** [gobanos/cargo-aoc#35]
I noticed `cargo aoc input` downloads inputs into directories by year.

Are there any plans to support multiple years per project in the runner?  It would be nice to have all my advent years in the same repo.

**Comment by gobanos** [comment]
The main issue is how symbols are exported by `aoc_lib`, in order to be used by `aoc_runner` and `aoc_generator` (which are year agnostic ATM).

**Comment by trebinor** [comment]
Makes sense.  Something to keep in mind for the future, if it's not too hard.

**Comment by RichoDemus** [comment]
This would be a great feature!

**Comment by kelvindecosta** [comment]
Hey @gobanos, I hope you're doing well!

I would love to use this tool this year and was hoping to do some of the previous years problems as well. 

How could one use `cargo-aoc` for multiple years, if each year is its own cargo workspace?

**Comment by markjfisher** [comment]
> Hey @gobanos, I hope you're doing well!
> 
> I would love to use this tool this year and was hoping to do some of the previous years problems as well.
> 
> How could one use `cargo-aoc` for multiple years, if each year is its own cargo workspace?

I too wish this was supported directly. I did create a hack last year to be able to have all my code in 1 repo.
See https://github.com/markjfisher/advent-of-code-rust/tree/master

I had to add the current year to the `members` values in Cargo.toml, and exclude previous years, keeping common/shared code in a separate directory and including that in the members entry too. If your puzzles are purely in the year folders, then this kind of works.

Then you can compile and run with `cargo run --bin aoc2023` or the year you have enabled in the members list.

I'm planning on coming back to doing it in rust this year.
